### ISSUE TITLE
Jupyter syntax highlighting

### ISSUE THREAD

**Issue by max-sixty** [prql/PyPrql#45]
As a follow-up to #44, I wonder whether it's possible to get syntax highlighting in the jupyter cells.

Currently it inherits from python:
<img width="158" alt="image" src="https://user-images.githubusercontent.com/5635139/169623470-6cb8131f-e23c-4994-acf2-be76eba6fff0.png">

There's a discussion [here](https://discourse.jupyter.org/t/getting-syntax-highlighting-to-work-for-custom-cell-magic/11734/5) on this, which links to https://github.com/jupyter-lsp/jupyterlab-lsp/pull/553, which looks good, and not _too_ involved, to get `%%bigquery` working. They have the advantage that there's already available syntax for SQL. I would hope that we could reuse one of our many existing definitions, but I've said that a few times before about syntax definitions :)

**Comment by rbpatt2019** [comment]
Perhaps it's just too early and I need more coffee, but the above approach isn't twigging for me. Do we need a separate typescript plugin to handle this?

**Comment by max-sixty** [comment]
Yes, unfortunately I think this will be in typescript. And I'm also not sure how the typescript file is linked to (does it rely on other code in `jupyterlab-lsp`??)...

**Comment by max-sixty** [comment]
I think this would be great to do, I may give this a swing soon. Unfortunately it involve JS, my great weakness...

Another reference: https://twitter.com/Mbussonn/status/636626829269442562 from @Carreau, suggests it should be possible

**Comment by max-sixty** [comment]
OK, this is a bigger project than I imagined. Here's the thread for Stan: https://github.com/jupyterlab/jupyterlab/issues/5504. So it's possible, but not easy, and hardly documented.

(and that's for jupyterlab, it's not completely clear that would work for the notebook. And it might be really hard for Colab: https://github.com/googlecolab/colabtools/issues/2039)

It looks like it depends on at least https://github.com/prql/prql/issues/1102

**Comment by max-sixty** [comment]
The Lezer grammar is coming along (thanks to @vanillajonathan ) — including packaging & tests.

So I'll take another look at doing this, despite my very poor JS.

Here's an example of a working extension: https://github.com/WardBrian/jupyterlab-stan-highlight/blob/master/lib/stan.js
